### DISCUSSION TITLE
Forgot Password url vs Microsoft safelinks

### DISCUSSION THREAD

**Discussion by mcardia** [keycloak/keycloak#19572] [Q&A]
Hi!

When we use the forgot password feature with a Microsoft hosted e-mail that uses the safelinks feature, the link breaks.
I found out that it is keycloak uses the | (pipe) character in the link.

Any clues about how to solve this from the keycloak side of the equation?

**Comment by darius-m** [comment]
Do the links work if you replace the pipe with the equivalent ASCII code (I think `%7C`)? If it does, I'd say it's a bug in the safelinks feature.
